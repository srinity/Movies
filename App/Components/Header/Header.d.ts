import { FC } from 'react'
import { TextProps } from 'react-native'

export interface HeaderProps extends TextProps {
  /**
   * The title of the header
   *
   */
  title?: string

  /**
   * Boolean value that controls whether to show the back button or not.
   *
   * @default false
   */
  withBack?: boolean

  /**
   * Method that will be invoked when back icon is pressed.
   *
   */
  onBackPress?(): void
}

export default class Header extends FC<HeaderProps> {}
