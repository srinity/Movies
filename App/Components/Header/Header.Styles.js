import { StyleSheet } from 'react-native'

import { Colors, Fonts } from '../../Theme'

export default StyleSheet.create({
  containerStyle: withBack => ({
    paddingVertical: 7,
    paddingHorizontal: 15,
    backgroundColor: Colors.backgroundColor,
    height: 60,
    flexDirection: 'row',
    ...(withBack ? { justifyContent: 'center' } : {}),
    // justifyContent: 'center',
    alignItems: 'center'
  }),
  titleStyle: {
    fontSize: Fonts.h1,
    fontWeight: 'bold'
  },
  backIconStyle: {
    position: 'absolute',
    left: 15,
    padding: 7
  }
})
