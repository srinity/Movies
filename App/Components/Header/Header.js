import React from 'react'
import { View, Text, TouchableWithoutFeedback } from 'react-native'
import PropTypes from 'prop-types'

import { Icon } from '../Icon'

import styles from './Header.Styles'

function Header({ withBack, title, onBackPress, ...props }) {
  return (
    <View style={styles.containerStyle(withBack)}>
      {withBack ? (
        <TouchableWithoutFeedback onPress={onBackPress}>
          <Icon
            type="entypo"
            name="chevron-small-left"
            size={35}
            style={styles.backIconStyle}
          />
        </TouchableWithoutFeedback>
      ) : null}

      {title ? (
        <Text style={styles.titleStyle} {...props}>
          {title}
        </Text>
      ) : null}
    </View>
  )
}

Header.propTypes = {
  withBack: PropTypes.bool,
  title: PropTypes.string,
  onBackPress: PropTypes.func
}

export default Header
