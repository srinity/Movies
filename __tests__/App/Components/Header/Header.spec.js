import React from 'react'
import renderer from 'react-test-renderer'

import Header from '../../../../App/Components/Header/Header'

describe('Header Component', () => {
  it('should render title correctly', () => {
    const header = renderer.create(<Header title="Movies" />)
    expect(header).toMatchSnapshot()
  })

  it('should render back button correctly', () => {
    const header = renderer.create(<Header withBack />)
    expect(header).toMatchSnapshot()
  })

  it('should render back button with title correctly', () => {
    const header = renderer.create(<Header title="Movies" withBack />)
    expect(header).toMatchSnapshot()
  })
})
