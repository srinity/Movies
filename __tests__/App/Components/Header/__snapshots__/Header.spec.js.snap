// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render back button correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 7,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 35,
        },
        Object {
          "left": 15,
          "padding": 7,
          "position": "absolute",
        },
        Object {
          "fontFamily": "Entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Header Component should render back button with title correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 7,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 35,
        },
        Object {
          "left": 15,
          "padding": 7,
          "position": "absolute",
        },
        Object {
          "fontFamily": "Entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Movies
  </Text>
</View>
`;

exports[`Header Component should render title correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flexDirection": "row",
      "height": 60,
      "paddingHorizontal": 15,
      "paddingVertical": 7,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Movies
  </Text>
</View>
`;
